### ISSUE TITLE
TextField toolbars (Material & Cupertino) do not show up when using nested MaterialApps

### ISSUE THREAD

**Issue by adriancmurray** [flutter/flutter#36124]
@justinmc - you seem to be the one tackling the issues concerning the toolbar. This should be easily reproducible and I'd wager there are other things that can cause this as well, perhaps it's similar/related to the FocusNode problem with trying to gain text focus while having a textfield in an overlay? 

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(new MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      home: new MaterialApp(
        home: new Scaffold(
          body: new SafeArea(
            child:new Column(
              children: <Widget>[
                new Expanded(child: Container(),),
                new Container(
                  color: Colors.grey,
                  child: new TextField(),
                )
              ],
            )
          ),
        ),
      ),
    );
  }
}

```

Flutter Doc : 
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.7.8+hotfix.3, on Mac OS X 10.14.4 18E226, locale en-US)
 
[!] Android toolchain - develop for Android devices (Android SDK version 27.0.3)
    ✗ Flutter requires Android SDK 28 and the Android BuildTools 28.0.3
      To update using sdkmanager, run:
        "/Users/Adrian/Library/Android/sdk/tools/bin/sdkmanager" "platforms;android-28" "build-tools;28.0.3"
      or visit https://flutter.dev/setup/#android-setup for detailed instructions.
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
[✓] iOS tools - develop for iOS devices
[✓] Android Studio (version 3.2)
[✓] VS Code (version 1.34.0)
[✓] Connected device (1 available)

! Doctor found issues in 1 category.

```

**Comment by LongCatIsLooong** [comment]
Fixed in https://github.com/flutter/flutter/pull/35219? I wasn't able to repro on master.

**Comment by adriancmurray** [comment]
I still had the same problems within my own app on master but I didn’t try the sample code on that build.

**Comment by justinmc** [comment]
I also can't reproduce this on master.  @adriancmurray Maybe there is something else going on in your code that exposes this besides the nested `MaterialApp`s?

**Comment by adriancmurray** [comment]
I'd figure the same thing, except that this simple code reproduces the issue on my device. I'll try again.

**Comment by adriancmurray** [comment]
Okay, it took a bit because I kept running into other problems. Long story short, I'm still seeing the error even with Flutter 1.8.2-pre.51

I created an entirely new app to test this, copied the simple code I supplied above and ran. Then I ran into this issue : https://github.com/flutter/flutter/issues/36150

Updated to the latest version of Android Studio and the Flutter plugin. This is the error I'm seeing now:

```
flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following assertion was thrown during performLayout():
flutter: 'package:flutter/src/rendering/box.dart': Failed assertion: line 1692 pos 18: 'debugDoingThisResize
flutter: || debugDoingThisLayout ||
flutter:                  (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent)': is not
flutter: true.
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
flutter:
flutter: User-created ancestor of the error-causing widget was:
flutter:   MaterialApp file:///Users/Adrian/dev/bucket/test_bucket/lib/main.dart:12:17
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #2      RenderBox.size.<anonymous closure> (package:flutter/src/rendering/box.dart:1692:18)
flutter: #3      RenderBox.size (package:flutter/src/rendering/box.dart:1698:6)
flutter: #4      RenderFractionalTranslation.applyPaintTransform (package:flutter/src/rendering/proxy_box.dart:2425:24)
flutter: #5      RenderObject.getTransformTo (package:flutter/src/rendering/object.dart:2187:24)
flutter: #6      RenderBox.globalToLocal (package:flutter/src/rendering/box.dart:2236:31)
flutter: #7      _ToolbarRenderBox.performLayout (package:flutter/src/cupertino/text_selection.dart:167:35)
flutter: #8      RenderObject.layout (package:flutter/src/rendering/object.dart:1639:7)
flutter: #9      _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #10     RenderObject.layout (package:flutter/src/rendering/object.dart:1639:7)
flutter: #11     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #12     RenderObject.layout (package:flutter/src/rendering/object.dart:1639:7)
flutter: #13     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
flutter: #14     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1516:7)
flutter: #15     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:783:18)
flutter: #16     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:346:19)
flutter: #17     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:702:13)
flutter: #18     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:285:5)
flutter: #19     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1033:15)
flutter: #20     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:975:9)
flutter: #21     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:891:5)
flutter: #25     _invoke (dart:ui/hooks.dart:236:10)
flutter: #26     _drawFrame (dart:ui/hooks.dart:194:3)
flutter: (elided 5 frames from class _AssertionError and package dart:async)
flutter:
flutter: The following RenderObject was being processed when the exception was fired: _ToolbarRenderBox#2a0c8 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE:
flutter:   needs compositing
flutter:   creator: CupertinoTextSelectionToolbar ← CompositedTransformFollower ← FadeTransition ←
flutter:     _OverlayEntry-[LabeledGlobalKey<_OverlayEntryState>#dd1aa] ← Stack ← _Theatre ←
flutter:     Overlay-[LabeledGlobalKey<OverlayState>#12170] ← _FocusMarker ← Semantics ← FocusScope ←
flutter:     AbsorbPointer ← Listener ← ⋯
flutter:   parentData: <none> (can use size)
flutter:   constraints: BoxConstraints(w=414.0, h=896.0)
flutter:   size: Size(414.0, 896.0)
flutter: This RenderObject had the following descendants (showing up to depth 5):
flutter:     child: RenderDecoratedBox#f2e5a relayoutBoundary=up1 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:       child: RenderFlex#5bda7 relayoutBoundary=up2 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:         child 1: RenderSemanticsGestureHandler#f6135 relayoutBoundary=up3 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:           child: RenderPointerListener#7e92d relayoutBoundary=up4 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:             child: RenderSemanticsAnnotations#094e4 relayoutBoundary=up5 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:         child 2: RenderConstrainedBox#20ab1 relayoutBoundary=up3 NEEDS-PAINT
flutter:         child 3: RenderSemanticsGestureHandler#8536c relayoutBoundary=up3 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:           child: RenderPointerListener#a9886 relayoutBoundary=up4 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:             child: RenderSemanticsAnnotations#9c17f relayoutBoundary=up5 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:         child 4: RenderConstrainedBox#156e4 relayoutBoundary=up3 NEEDS-PAINT
flutter:         child 5: RenderSemanticsGestureHandler#201c8 relayoutBoundary=up3 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:           child: RenderPointerListener#9ed20 relayoutBoundary=up4 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter:             child: RenderSemanticsAnnotations#1c455 relayoutBoundary=up5 NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: NoSuchMethodError: The method 'toDouble' was called on null.

```

**Comment by adriancmurray** [comment]
[FIXED - sort of] I've been testing in debug mode the entire time. I decided to test in profile mode since I couldn't figure out why either of you couldn't see the issue (I also started testing on the emulator and was about to get another kind of iOS device). Apparently this is only a problem in debug mode as I found profile mode ran the code above just fine. If you think that this isn't a problem then we can mark this as closed. Thank you.
